roles/kubevirt_web_ui/tasks/required_params.yml

```diff
@@ -0,0 +1,38 @@
+---
+#### Verify required variables
+
+- name: Verify required kubevirt_web_ui_image_name variable is set
+  fail:
+    msg: kubevirt_web_ui_image_name must be set to deploy kubevirt-web-ui
+  when: kubevirt_web_ui_image_name is not defined
+
+#### Autodiscovery of missing variables
+# Assumptions for autodiscovery:
+# - console's namespace is hardcoded to "openshift-console" (as done in openshift-ansible)
+# - the openshift console is mandatory part of OKD installation
+
+- name: Discover public_master_hostname from openshift console deployment
+  shell: "{{ openshift_client_binary }} get configmap console-config -n openshift-console -o jsonpath='{.data}' | grep 'masterPublicURL:' | sed 's/^.*masterPublicURL: *http.*:\\/\\///g'"
+  register: parsedMasterPublicURL
+  when: public_master_hostname | default("") | string == ""
+
```

Comment:
can any of these be defaults? Or are they dynamic in the sense that a static default won't always be accurate?

Reply:
I don't know about any reasonable static default for these two variables (`public_master_hostname` or `openshift_master_default_subdomain`). The default in examples states sort of `foo.bar.example.com`. 

Recent implementation either
- gathers the values from console's deployment (should be nearly 100% cases, if I don't miss something)
- or fails on undefined `public_master_hostname` or `openshift_master_default_subdomain`

I can add check later to this file for the `undefined` value of these two facts so we fail early and with more explanatory message (means pointing to documentation). But this flow is not expected to happen and is covered in documentation. WDYT?

Reply:
My thinking is that these vars can be set as required vars and we fail without them being set instead of going out to find them.   But, if you think that the configmap will always be there and it will always use the same names to look up these vars, then it could work.

Reply:
In my understanding, the configmap, resp. `openshift-console`, will be there as it is (recently) mandatory part of an openshift 3.11 installation. The key names of the configmap are not expected to be changed but we can keep monitoring such potential change as the kubevirt/web-ui is fork of the openshift/console and we regularly keep it in sync.
So if we can gather these values automatically, I would not delegate that on the user.

If the assumption above is not met, the execution is expected to fail instead of passing corrupted values.